platform/v06/platform.go

```diff
@@ -1,17 +1,13 @@
 package v06
 
+import v05 "github.com/buildpacks/lifecycle/platform/v05"
+
 type Platform struct {
-	api string
+	*v05.Platform
 }
 
 func NewPlatform(apiStr string) *Platform {
-	return &Platform{api: apiStr}
-}
-
-func (p *Platform) API() string {
-	return p.api
-}
-
-func (p *Platform) SupportsAssetPackages() bool {
-	return false
+	return &Platform{
+		Platform: v05.NewPlatform(apiStr),
```

Comment:
I am pretty sure that I wrote the original code, but I find it weird that the constructor in the `v05` package even takes a string as an argument. Shouldn't it always be `"0.5"`?

Separately though, I think we might start to get confused if we had a `v05.Platform` that returned `"0.6"` for its API... thinking...

Reply:
Here is what I was thinking of: https://play.golang.org/p/adQu0qhSHej

Reply:
I feel like... it works, but maybe it's too confusing and we're better off copying all of the methods to each platform. Curious to know what others think!

Reply:
Or maybe instead of silently delegating to the previous platform, we have a `previousPlatform` field and a method that explicitly calls the method on the previous platform, like:
```
func (p *Platform) CodeFor(errType cmd.LifecycleExitError) int {
	return p.previousPlatform.CodeFor(errType)
}
```

Reply:
See other discussion: https://github.com/buildpacks/lifecycle/pull/591#discussion_r635433290